### PR TITLE
doc/user: spell out how to file support ticket

### PR DESCRIPTION
**Pull request by benesch**
### Motivation

  * This PR adds known-desirable documentation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

File: doc/user/content/support.md [hunk 1]
```diff
@@ -9,9 +9,40 @@ menu:
...
 
-If you're a paying customer, you may also have access to premium support via a shared channel within your company's Slack or Microsoft Teams workspace.
+### Customers
+
+If you're a paying customer, you can access Materialize support via a shared
+channel within your company's Slack or Microsoft Teams workspace.
+
+To file a support ticket:
+
+1. Send a message to your shared channel describing your issue.
```
Comment (morsapaes) [inline doc/user/content/support.md:23]:
```suggestion
1. Send a message to the shared channel describing your issue.
```

File: doc/user/content/support.md [hunk 2]
```diff
@@ -9,9 +9,40 @@ menu:
...
+2. Apply the `:ticket:` (🎫) emoji to your message.
+3. Click the **File a ticket** button that appears.
+4. Select the appropriate priority for your ticket. Consult your support
+   contract to determine the definition of each priority level and the
+   associated guaranteed response time.
```
Comment (morsapaes) [inline doc/user/content/support.md:28]:
```suggestion
4. Select the appropriate priority for your ticket. You can find the definition
    of each priority level and the associated guaranteed response time in your
    support contract.
```

File: doc/user/content/support.md [hunk 3]
```diff
@@ -9,9 +9,40 @@ menu:
...
+5. Materialize will respond to the ticket within the response time outlined
+   in your support contract.
```
Comment (morsapaes) [inline doc/user/content/support.md:30]:
```suggestion
5. The Materialize team will respond to the ticket within the response time
   associated with its priority.
```

**Review by morsapaes — APPROVED** [review]
Thanks for working on improving this process, @benesch.

File: doc/user/content/support.md (hunk 2 above)
Comment (benesch) [reply doc/user/content/support.md]: Refactored away to refer to a table in this document.

**Comment by benesch** [comment]
I've updated this with a description of the priorities and how they map to error severities that are spelled out in folks' support contracts.

**Review by sjwiesman — APPROVED** [review]

**Pull request merged**